api/objects.proto

```diff
@@ -75,6 +75,10 @@ message Node {
 	// shows the privilege level that the CA would currently grant when
 	// issuing or renewing the node's certificate.
 	NodeRole role = 9;
+
+	// LBAttachments carry per-network node attachment required for
+	// optimally placed distributed load-balancers.
+	map<string, NetworkAttachment> lb_attachments = 10;
```

Comment:
This needs to operate cooperatively with the `attachment` field. Do we need to deprecate that field?

Reply:
attachment can be deprecated.  How do I deprecate it?  Can you point me to an example?  Thanks.

Reply:
There is a GRPC directive that can mark it as deprecated. We'll have to look more into how this field appears in the Docker API to see impact there. (Check the swagger docs)

Also, this should not be a map type. Declare a new message with the decorated attributes.

Reply:
Will do. Thanks for the feedback.

Reply:
Regarding deprecating attachment:  I followed the other examples in api\*.proto and marked the field at deprecated using [deprecated=true].  
Note that this differs from the recommendation from protobuf documentation https://developers.google.com/protocol-buffers/docs/proto3#updating
but, it is consistent with other deprecated fields in swarmkit.

Reply:
Regarding "Also, this should not be a map type. Declare a new message with the decorated attributes."
Thought about this, tried it out, and it looks really odd doing that.   I moved the map into it's own message and the map was the only field.
Why do you think the map should be a new message?   We use maps in many other messages.
Maybe I misunderstood your comment.
For now, I left it as is, and updated the documentation as suggested by aaronlehmann.

Reply:
@pradipd Maps are almost never the right data structure, unless there is a uniqueness constraint (and even then, I would recommend against their usage). It also makes it unclear what the value of the key is.

Using a proper structure makes it clear what the values actually are. However, I am pretty confused about why you have it this way, since the network id is already a part of the attachment. This looks completely redundant.

Reply:
The redundancy is intentional.  Maps are useful for scalability.  I opted for a map (as opposed to an array) to make the lookup's more efficient.  E.g. in AllocateLBAttachment().  
If we use an array then we will iterate over all the networks for every node we allocate.  
My assumptions is that there will be multiple networks.  But, perhaps we will always have a very small number of networks (like 1 or 2).
I can change it to an array if you feel strongly about it.

Reply:
@pradipd So, index the array in the implementation. Don't choose the datastructure based on the current implementation but on what it represents.

Reply:
Addressed in the latest changes.